### PR TITLE
Guard against removing other user's authenticator

### PR DESCRIPTION
**Pull request by IniZio**
ref DEV-1153

### CODE REVIEW

File: pkg/lib/interaction/nodes/remove_authenticator.go [hunk 1]
```diff
@@ -35,6 +36,14 @@ func (e *EdgeRemoveAuthenticator) Instantiate(ctx *interaction.Context, graph *i
 		return nil, err
 	}
 
+	if info.UserID != graph.MustGetUserID() {
```
Comment (tung2744) [inline pkg/lib/interaction/nodes/remove_authenticator.go:39]: Prefer to check in authenticator.Service

Comment (IniZio) [reply pkg/lib/interaction/nodes/remove_authenticator.go:39]: Authenticator facade is used by other transport e.g. userimport. In those cases deleting from other/no user id should be legitimate?

Comment (tung2744) [reply pkg/lib/interaction/nodes/remove_authenticator.go:39]: I see.
BTW, is there same problem in delete identity?

Comment (IniZio) [reply pkg/lib/interaction/nodes/remove_authenticator.go:39]: Found have similar issue in remove and update identity as well. Let me update that thanks!

File: pkg/lib/interaction/nodes/do_update_identity.go [hunk 2]
```diff
@@ -46,6 +46,14 @@ func (n *NodeDoUpdateIdentity) Prepare(ctx *interaction.Context, graph *interact
...
 	return []interaction.Effect{
 		interaction.EffectRun(func(ctx *interaction.Context, graph *interaction.Graph, nodeIndex int) error {
+			if n.IdentityBeforeUpdate.UserID != graph.MustGetUserID() {
+				return api.NewInvariantViolated(
+					"IdentityNotBelongToUser",
+					"identity does not belong to the user",
+					nil,
+				)
+			}
+
```
Comment (louischan-oursky) [inline pkg/lib/interaction/nodes/do_update_identity.go:56]: This checking should be done in update_identity_end.go:22

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**